crates/compiler/builtins/roc/List.roc

```diff
@@ -369,30 +369,43 @@ contains = \list, needle ->
 ## You can use it in a pipeline:
 ##
 ##     [2, 4, 8]
-##         |> List.walk { start: 0, step: Num.add }
+##         |> List.walk 0 Num.add
 ##
 ## This returns 14 because:
-## * `state` starts at 0 (because of `start: 0`)
+## * `state` starts at 0
 ## * Each `step` runs `Num.add state elem`, and the return value becomes the new `state`.
 ##
 ## Here is a table of how `state` changes as [List.walk] walks over the elements
-## `[2, 4, 8]` using #Num.add as its `step` function to determine the next `state`.
-##
-## `state` | `elem` | `step state elem` (`Num.add state elem`)
-## --------+--------+-----------------------------------------
-## 0       |        |
-## 0       | 2      | 2
-## 2       | 4      | 6
-## 6       | 8      | 14
-##
-## So `state` goes through these changes:
-## 1. `0` (because of `start: 0`)
-## 2. `1` (because of `Num.add state elem` with `state` = 0 and `elem` = 1
+## `[2, 4, 8]` using [Num.add] as its `step` function to determine the next `state`.
+##
+## <table style="text-align: center">
+##   <thead>
+##     <tr>
+##       <th style="text-align: center; width: 5em">state</th>
+##       <th style="text-align: center; width: 5em">elem</th>
+##       <th style="text-align: center; width: 10em">Num.add state elem</th>
+##     </tr>
+##   </thead>
+##   <tbody>
+##     <tr>
+##       <td>0</td>
+##     </tr>
+##     <tr>
+##       <td>0</td> <td>2</td> <td>2</td>
+##     </tr>
+##     <tr>
+##       <td>2</td> <td>4</td> <td>6</td>
+##     </tr>
+##     <tr>
+##       <td>6</td> <td>8</td> <td>14</td>
+##     </tr>
+##   </tbody>
+## </table>
```

Comment:
This works at the moment, but I'd actually like for us not to use inline HTML in our docs. 😅 

The reason is that in the future we'll need to render these docs in the editor as well, and I really don't want us to have to support an entire HTML engine in the editor!

Instead, how about just making the previous table embedded in a code block so the columns line up?